launch/launch/substitutions/python_expression.py

```diff
@@ -47,15 +50,15 @@ def __init__(self, expression: SomeSubstitutionsType) -> None:
             'expression',
             'PythonExpression')
 
-        from ..utilities import normalize_to_list_of_substitutions
         self.__expression = normalize_to_list_of_substitutions(expression)
 
     @classmethod
     def parse(cls, data: Iterable[SomeSubstitutionsType]):
         """Parse `PythonExpression` substitution."""
-        if len(data) != 1:
-            raise TypeError('eval substitution expects 1 argument')
-        return cls, {'expression': data[0]}
+        if len(data) == 1:
```

Comment:
@kenji-miyake we should still catch `len(data) == 0` and raise an exception.

Reply:
Fixed in 205e7ccc04577d75098538ec916aae419ca58926.
And added a test.